packages/loader/container-loader/src/utils.ts

```diff
@@ -35,7 +35,12 @@ export interface IParsedUrl {
 	version: string | null | undefined;
 }
 
-export function parseUrl(url: string): IParsedUrl | undefined {
+/**
+ * Utility api to parse the IResolvedUrl.url into specific parts like querystring, path to get
+ * deep link info etc.
+ * @param url - This is the IResolvedUrl.url part of the resolved url.
+ */
+export function parseIResolvedUrlUrlIntoParts(url: string): IParsedUrl | undefined {
```

Comment:
Another name to consider

```suggestion
export function tryParseCompatibleResolvedUrl(url: string): IParsedUrl | undefined {
```

Having "try" in there and speaking of "compatible" resolved URLs helps give a clue that if you don't know, it might not work with your driver / resolver.